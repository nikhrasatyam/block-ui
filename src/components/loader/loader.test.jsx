import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import Enzyme from 'enzyme';
import Loader from './loader';
import UiStore from '../../stores/ui.store';


describe('Loader Test Cases', () => {
  let uiStore;

  beforeEach(() => {
    uiStore = new UiStore();
  });
  afterEach(() => { uiStore = null; });
  test('should render loader when showLoader is falsy', () => {
    const tree = renderer.create(<Provider uiStore={uiStore}><Loader /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render loader when showLoader is truthy', () => {
    uiStore.setShowLoader(true);
    const tree = renderer.create(<Provider uiStore={uiStore}><Loader /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should have Icon when showLoader is truthy', () => {
    uiStore.setShowLoader(true);
    const wrapper = Enzyme.mount(<Provider uiStore={uiStore}><Loader /></Provider>);
    expect(wrapper.find('Icon')).toHaveLength(1);
  });
});
