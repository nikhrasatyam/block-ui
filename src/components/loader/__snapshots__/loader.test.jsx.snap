// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Test Cases should render loader when showLoader is falsy 1`] = `null`;

exports[`Loader Test Cases should render loader when showLoader is truthy 1`] = `
<div
  className="sc-bdVaJa kuxarL"
>
  <i
    aria-hidden="true"
    className="circle notched loading icon"
    onClick={[Function]}
  />
</div>
`;

exports[`Loader Test Cases should render loader when showLoader is truthy 2`] = `
<div
  className="sc-bdVaJa kuxarL"
>
  <i
    aria-hidden="true"
    className="circle notched loading icon"
    onClick={[Function]}
  />
</div>
`;
